docs/designers-developers/developers/tutorials/create-block/author-experience.md

```diff
@@ -0,0 +1,129 @@
+
+## Placeholder
+
+The state when a block has been inserted, but no data has been entered yet, is called a placeholder. There is a `Placeholder` component built that gives us a standard look. You can see example placeholders in use with the image and embed blocks.
+
+To use the Placeholder, wrap the `<TextControl>` component so it becomes a child element of the `<Placeholder>` component. Try it out in your code. After updating, you might have something like:
+
+```jsx
+import { Placeholder, TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+export default function Edit( { attributes, className, setAttributes } ) {
+    return (
+        <div className={ className }>
+			<Placeholder
+			  label="Gutenpride Block"
+			  instructions="Add your message"
+			>
+			  <TextControl
+			    value={ attributes.message }
+			    onChange={ ( val ) => setAttributes( { message: val } ) }
+			  />
+			</Placeholder>
+        </div>
+    );
+}
+```
+
+## isSelected Ternary Function
+
+The placeholder looks ok, for a simple text message it may or may not be what you are looking for. However, the placeholder can be useful if you are replacing the block after what is typed in, similar to the embed blocks.
+
+For this we can use a ternary function, to display content based on a value being set or not. A ternary function is an inline if-else statement, using the syntax:
+
+```js
+  ( clause ) ? ( doIfTrue ) : ( doIfFalse )
+```
+
+This can be used inside a block to control what shows when a parameter is set or not. A simple case that checks if the `message` is set might look like:
+
+```jsx
+  return (
+    <div>
+      { attributes.message ?
+        <div> Message: { attributes.message }</div> :
+        <div> No Message <TextField/> </div>
+      }
+  );
+```
+
+If we only used the above check, as soon as we type anything, the textfield would disappear since the message would be set. So we need to pair with the `isSelected` parameter.
+
+The `isSelected` parameter is passed in to the `edit` function and is set to true if the block is selected in the editor (currently editing) otherwise set to false (editing elsewhere).
+
+Using that parameter, we can use the logic:
+
+```js
+attributes.message && ! isSelected
+```
+
+If the message is set and `!isSelected`, meaning we are not editing the block, the focus is elsewhere, then display the message not the text field.
+
+All so this combined together here's what the edit function looks like this:
+
+```jsx
+import { Placeholder, TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+export default function Edit( { attributes, className, isSelected, setAttributes } ) {
+    return (
+        <div>
+        { attributes.message && !isSelected ?
+            <div className={ className }>
```

Comment:
This class name was previously applied to the root `div`, was there any particular reason to change it? Now, placeholder state doesn’t use this class.

Reply:
I don't think so, I moved it back to root element. I think it just happened while iterating.